### DISCUSSION TITLE
Prompt Engineering 一個基於大型語言模型誕生的工程科學

### DISCUSSION THREAD

**Discussion by JaYanlloz** [1101-GS4719-C/Python-Community-GS4719-C#2199] [科技新知]
伴隨著大型語言模型的快速發展與普及，Prompt Engineering因應而生，成為生成式AI領域中非常關鍵的技術。
其通過設計與優化模型輸入（即提示，prompt），來引導AI生成高質量、精確且相關的輸出。
Prompt的設計決定了模型的理解能力及其生成結果的準確性與一致性，因此在文本生成、問題解答、內容創作等領域的LLMs發揮著重要作用。

我認為這門技術的核心技術在於了解如何通過精準的指令，最大化語言模型的潛力，並解決特定問題。透過不同的提示設計，AI模型能夠執行各種任務，包括文本摘要、翻譯、對話生成，甚至是程式碼生成等應用。

而且隨著生成式AI越來越廣泛的應用，Prompt Engineering的重要性也逐漸提升，成為了訓練和運用 AI 模型不可或缺的部分。它不僅需要理解語言模型的工作原理，也需要掌握如何有效與模型互動，以達成期望的結果。

我認為這符合目前主流大型公司認為的下一個世代的程式語言，即以自然語言就可以進行programing。

雖然有蠻多聲音認為Prompt Engineering在不久的將來會被Autoprompt取代，但我仍認為他會是一個非常值得研究的「工程學」
但有一說一，這東西真的有點玄...，好了，我要去做第三個月的第n次prompt調整了...

**Comment by Tchang323** [comment]
在AI浪潮的時代，懂得如何精確下prompt真的很重要，讓模型能夠更貼近使用者想要的生成結果。有些生成式AI模型在訓練微調的過程，不只是要下prompt，也會需要設置Captioin。可以看出Prompt engineering的重要性(我們下周有一堂課的演講主題就是這個)，現在網路上很多人都在教學如何下prompt，我也認同你的觀點，覺得這個是非常值得研究的一項學問。

**Comment by Chattenoire** [comment]
雖然Zero-shot, One-shot, Template等等Prompt engineering是大家一直強調要運用的，但有時候用AI是因為沒那麼多時間自己檢索資料；要把這些模板打出來就要耗費一段時間了......

另外，其實也有奇怪的prompt engineering paper，像是底下這篇:
https://arxiv.org/abs/2201.11903

**Comment by 112121py** [comment]
這東西是真的玄，之前用過coze下prompt(雖然我不確定那個算不算)，要對bot下一個精確的prompt真的還蠻困難的，同一個意思換句話說，他的表現就完全不一樣，超搞，但下過幾次習慣後會抓到一點點技巧，雖然有時候還是會大失敗，但這是個蠻值得研究一下的技術(?
